edb/pgsql/compiler/clauses.py

```diff
@@ -489,7 +489,7 @@ def scan_check_ctes(
             name='flag', val=pgast.BooleanConstant(val=True)
         )],
         relation=pgast.RelRangeVar(relation=pgast.Relation(
-            schemaname='edgedb', name='_dml_dummy')),
+            name='_dml_dummy')),
```

Comment:
Do you think we should try to invalidate the persistent cache because of this change, or should we just instruct affected users to clear the cache manually?

(This is a more general question about miscompilation bugs and the persistent cache that I'm not sure we've ever properly figured out)

Reply:
Oh true! One way that might work is to always "upgrade" the persistent cache automatically after server upgrade (by recompiling the cached queries) even if there are no compiler changes - if it's already in downtime maintenance, why not make the most of it and make sure the query cache is up to date. Let me draft a PR for that.

> should we just instruct affected users to clear the cache manually?

I'd use that as the last resort when we failed to make the server fix itself (which is likely the case now, but let me fix that).

Reply:
Is extending the downtime (possibly by a lot) better than a cache flush? It's not totally obvious to me.

Maybe it would be best to come up with an empty cache and then repopulate in the background?

Reply:
I don't consider cache recompilation to be a release blocker; it's been an outstanding issue through all of 5.0, so we can live with it a little longer.